roles/network_plugin/calico/tasks/install.yml

```diff
@@ -331,11 +332,11 @@
       {"apiVersion": "projectcalico.org/v3",
       "kind": "BGPPeer",
       "metadata": {
-        "name": "{{ inventory_hostname }}-{{ item.router_id }}"
+        "name": "{{ item.node }}-{{ item.router_id }}"
       },
       "spec": {
         "asNumber": "{{ item.as }}",
-        "node": "{{ inventory_hostname }}",
+        "node": "{{ item.node }}",
```

Comment:
`spec.node` is supposed to match the node's the value of `metadata.name` as defined in the previous task (which takes the value of `inventory_hostname`).
There is no need to define a "node" key in `peers` items.
`peers` should be defined in hostvars as it is a "per node peering"

Reply:
@mirwan  The above lack of nodes being added had all the peers for calico being added, so half the routes were failing.  Should only be adding the per-node.  I did fix for the generic case.

Reply:
I think this is because you defined `peers` in groupvars whereas it should be defined in hostvars (where each `peers` list is different from one to another) am I right ?

Reply:
The peers in the examples are defined in the k8s-net-calico.yml in the group_vars/k8s-cluster/ directory.  I followed this example.  https://gitlab.staging.cncf.ci/cncf/kubespray/tree/master/docs/calico_peer_example
but added nodes.

Also not handled are nodeSelectors.